MachineArithmetic/Z3AST.class.st

```diff
@@ -19,6 +19,7 @@ Z3AST class >> fromExternalAddress: anExternalAddress inContext: aZ3Context [
 	| kind sort |
 
 	self assert: self == Z3AST description: 'Attempting to instantiate AST subclass directly'.
+	anExternalAddress isNull ifTrue: [ ^nil ].
```

Comment:
I do agree having this check is good. However:

  * I suggest to implement the check as `anExternalAddress asInteger == 0` rather than using `#isNull`. This should work on St/X too. 
  * Are there cases where we a NULL pointer that are "legal" - I mean it's not a manifestation of a problem somewhere else? If not, I'd rather see `self assert: anExternalAddress asInteger ~~ 0` than returning `nil`.

Reply:
Yes, there are: `Z3 class>>query_constructor:...` is one of them.